sdk/azidentity/live_test.go

```diff
@@ -89,21 +88,10 @@ func init() {
 func TestMain(m *testing.M) {
 	switch recording.GetRecordMode() {
 	case recording.PlaybackMode:
-		// enable BodilessMatcher because we don't record request bodies
-		// TODO: add an API for this to sdk/internal
-		req, err := http.NewRequest("POST", "http://localhost:5000/Admin/SetMatcher", http.NoBody)
+		err := recording.SetBodilessMatcher(nil, nil)
 		if err != nil {
 			panic(err)
 		}
-		req.Header["x-abstraction-identifier"] = []string{"BodilessMatcher"}
-		res, err := http.DefaultClient.Do(req)
-		if err != nil {
-			panic(err)
-		}
-		if res.StatusCode != http.StatusOK {
-			log.Panicf("failed to enable BodilessMatcher: %v", res)
-		}
-		// TODO: reset matcher
```

Comment:
Is this unnecessary now? My concern is that playing back azidentity tests leaves body matching disabled. If I then run, say, Key Vault tests, and don't restart the proxy first, body matching is still disabled, and tests that should fail because of the bug I'm adding may pass instead.

Reply:
In KV I have a `ResetProxy` at the beginning of `TestMain` that way I'm always running from a fresh proxy. Do you want me to add that to identity as well? It's already there on RecordingMode, adding to PlaybackMode makes sense as well.

Reply:
Well, it depends on what all `ResetProxy` does. I want the proxy to be in its default state before and after azidentity tests. I don't want azidentity tests to disable the default OAuth response sanitizer; that creates a risk of recording secrets for another package.

Reply:
`ResetProxy` returns the proxy to it's default state, I believe this is what you want.

Reply:
In that case I think this function should call `ResetProxy` before and after every recording or playback run.

Reply:
Yes, that's why I have at the beginning of [`TestMain`](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/keyvault/azkeys/utils_test.go#L35-L38) in the key vault test libraries.

Reply:
I added the reset after test runs as well.

Reply:
I'm confused. It looks like you're enabling the bodiless matcher after the run? I want to restore the proxy's default settings i.e. to enable the default matcher and any default sanitizers. I want the proxy in its default state at the start and end of `TestMain`, when not in live mode. I don't want to do anything with the proxy in live mode.

Also, looking at the code, I see that past me believed `ResetProxy` removes default sanitizers--if the OAuth sanitizer is a default, that must be correct because otherwise these recordings wouldn't work. Is there an easy way to restore proxy defaults?

Reply:
If you call `ResetProxy`, the OAuth sanitizer will be on. Confirmed by @scbedd . ResetProxy restores to defaults.

Reply:
Yep! Hitting Admin/Reset gets directed to [here](https://github.com/Azure/azure-sdk-tools/blob/55f1b2f5eb63e1bd373be66e685c6fe9d61e0698/tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs#L437)

The default constructor (which is called for our singleton instance of RecordingHandler at server startup) calls SetDefaultExtensions(). 

But...is oauth sanitizer a default?

Reply:
> If you call `ResetProxy`, the OAuth sanitizer will be on. Confirmed by @scbedd . ResetProxy restores to defaults.

Great. About the block after `m.Run()` then--why enable the bodiless matcher there?

> But...is oauth sanitizer a default?

I hope so. Otherwise, the proxy would record authentication secrets by default.

Reply:
After the m.Run should be a Reset, I think that was a copy-pasta error